mypy/types.py

```diff
@@ -1546,6 +1549,57 @@ def keywords_str(self, a: Iterable[Tuple[str, Type]]) -> str:
         ])
 
 
+class PrettyTypeStrVisitor(TypeStrVisitor):
+    """Overloaded Visitor for pretty-printing types into user-faceing strings.
+
+    Attempts to (but usually doesnt) match original formatting
+
+    Notes:
+     - Represent unbound types as Foo? or Foo?[...].
+     - Represent the NoneTyp type as None.
+    """
+    def visit_none_type(self, t: NoneTyp) -> str:
+        # Note: This isn't discernable from the None value (for readiblity)
+        return "None"
+
+    def visit_callable_type(self, t: CallableType) -> str:
+        s = ''
+        bare_asterisk = False
+        for i in range(len(t.arg_types)):
+            if s != '':
+                s += ', '
+            if t.arg_kinds[i] in (ARG_NAMED, ARG_NAMED_OPT) and not bare_asterisk:
+                s += '*, '
+                bare_asterisk = True
+            if t.arg_kinds[i] == ARG_STAR:
+                s += '*'
+            if t.arg_kinds[i] == ARG_STAR2:
+                s += '**'
+            name = t.arg_names[i]
+            if name:
+                s += name + ': '
+            s += t.arg_types[i].accept(self)
+            if t.arg_kinds[i] in (ARG_OPT, ARG_NAMED_OPT):
+                s += ' ='
+
+        if t.definition is not None and t.definition.name() is not None:
+            # If we got a "special arg" (i.e: self, cls, etc...), prepend it to the arg list
+            if t.arg_names != t.definition.arg_names and len(t.definition.arg_names) > 0:
```

Comment:
I'm trying to get special args like `self` or `cls` here, but since they're not in `t.arg_names` I have to reach through to `t.definiton.arg_names` and do a comparison.  This this the most effective way to do this?